### PR TITLE
Added registration and updated views.

### PR DESCRIPTION
**Pull request by SoroorAghamiri**
#68

### CODE REVIEW

File: src/main/java/ca/concordia/soen6011/controller/RegistrationController.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        employer.setEmail(email);
+        employer.setFirstname(firstName);
+        employer.setLastname(lastName);
+
+        employer.setPassword(password);
+        //set user type
+        employerRepository.save(employer);
+
+    }
+
```
Comment (sepehr9877) [inline src/main/java/ca/concordia/soen6011/controller/RegistrationController.java:92]: @SoroorAghamiri ,it is good thank you , just one thing left to complete this functionality. that is authentication , so you are going to have to create an annotation ,as we put this annotation on those controller to limit user access. So once you  create this custom annotation , then you have to use AOP concept , to run a function for doing authentication before calling that annotation .
let me give you an example:
@MyCustomAnnotation 
@GetMapping("/admin)
public String mycontroller{......} 
then there is AOP concept here , you need to create a pointcut specifically for this annotation and then by using @Before on AOP function you are going to do the authentication thing . by doing so , we can easily put this annotation on every controller we want to put access limitation on .
Let me know if it is confusing or anything :)))

**Review by sepehr9877 — APPROVED** [review]

File: src/main/resources/templates/index.html [hunk 2]
```diff
@@ -10,7 +10,7 @@
 <div class="container">
   <h1>Whatever Job You Choose</h1>
 
-  <br><a href="\login\">Login</a>
+  <br><a href="\login">Login</a>
```
Comment (MajorTreble) [inline src/main/resources/templates/index.html:13]: thank you

File: src/main/java/ca/concordia/soen6011/controller/RegistrationController.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        model.addAttribute("user", new User());
+        return "registration_page"; }
+
+    @PostMapping("/register")
+    public ModelAndView register(@RequestParam("userName") String userName,
+                              @RequestParam("email") String email,
+                              @RequestParam("firstName") String firstName,
+                              @RequestParam("lastName") String lastName,
+                              @RequestParam("password") String password,
+                              @RequestParam("confirmPassword") String confirmPassword,
```
Comment (MajorTreble) [inline src/main/java/ca/concordia/soen6011/controller/RegistrationController.java:38]: This is for form validation before submit. You don't need it here.

File: src/main/java/ca/concordia/soen6011/controller/RegistrationController.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+                              @RequestParam("type") String type,
+                                 Model model) {
+        User userAvailable = userRepository.findByUsernameAndPassword(userName,password);
```
Comment (MajorTreble) [inline src/main/java/ca/concordia/soen6011/controller/RegistrationController.java:41]: Should you use existsByUsername? Password doesn't matter.

File: src/main/java/ca/concordia/soen6011/controller/RegistrationController.java [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        ModelAndView modelAndView = new ModelAndView();
+        modelAndView.setViewName("registration_page");
+        if(userAvailable != null){
```
Comment (MajorTreble) [inline src/main/java/ca/concordia/soen6011/controller/RegistrationController.java:44]: Is there a way to do this on the form before submitting?

File: src/main/java/ca/concordia/soen6011/controller/RegistrationController.java [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+            //show error and ask if user wants to login
+            model.addAttribute("userExists", true);
+            return modelAndView;
+        }else{
+            boolean userNameUsed = userRepository.existsByUsername(userName);
```
Comment (MajorTreble) [inline src/main/java/ca/concordia/soen6011/controller/RegistrationController.java:49]: Why are you checking twice?

File: src/main/java/ca/concordia/soen6011/controller/RegistrationController.java [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+        Employer employer = new Employer();
+
+        employer.setUsername(username);
+        employer.setEmail(email);
+        employer.setFirstname(firstName);
+        employer.setLastname(lastName);
+
+        employer.setPassword(password);
+        //set user type
+        employerRepository.save(employer);
```
Comment (MajorTreble) [inline src/main/java/ca/concordia/soen6011/controller/RegistrationController.java:89]: Can there be an error?

File: src/main/java/ca/concordia/soen6011/controller/RegistrationController.java [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+        candidate.setEmail(email);
+        candidate.setFirstname(firstName);
+        candidate.setLastname(lastName);
+
+        candidate.setPassword(password);
+        //set user type
+        candidateRepository.save(candidate);
+    }
+
+    private boolean validateEmail(String email){
```
Comment (MajorTreble) [inline src/main/java/ca/concordia/soen6011/controller/RegistrationController.java:106]: Can you do this on the form?

**Review by MajorTreble — APPROVED** [review]

File: src/main/java/ca/concordia/soen6011/controller/RegistrationController.java (hunk 1 above)
Comment (SoroorAghamiri) [reply src/main/java/ca/concordia/soen6011/controller/RegistrationController.java:92]: I'll add that. Thank you!

File: src/main/java/ca/concordia/soen6011/controller/RegistrationController.java (hunk 5 above)
Comment (SoroorAghamiri) [reply src/main/java/ca/concordia/soen6011/controller/RegistrationController.java:44]: That's actually a good idea! I'll try that!

File: src/main/java/ca/concordia/soen6011/controller/RegistrationController.java (hunk 7 above)
Comment (SoroorAghamiri) [reply src/main/java/ca/concordia/soen6011/controller/RegistrationController.java:89]: I'll put it in a try catch block.

**Pull request merged**